src/main/java/com/redhat/devtools/lsp4ij/client/features/LSPCompletionProposal.java

```diff
@@ -155,12 +176,57 @@ private void updateCompletionItemFromResolved() {
         this.item = resolved;
     }
 
+    @NotNull
+    private String adjustSnippetContent(@NotNull String snippetContent) {
+        // If configured not to use a template for snippets that look like invocation arguments, simplify the snippet
+        if (!completionFeature.useTemplateForInvocationOnlySnippet(this.file)) {
+            Matcher invocationArgsSnippetMatcher = INVOCATION_ARGS_SNIPPET_PATTERN.matcher(snippetContent);
+            if (invocationArgsSnippetMatcher.find()) {
+                int invocationArgsStartIndex = invocationArgsSnippetMatcher.start();
+                int invocationArgsEndIndex = invocationArgsSnippetMatcher.end();
+
+                // Make sure that there are no other snippet variables aside from end outside of the invocation arguments
+                Matcher snippetVariableMatcher = SNIPPET_VARIABLE_PATTERN.matcher(snippetContent);
+                while (snippetVariableMatcher.find()) {
+                    if ((snippetVariableMatcher.start() < invocationArgsStartIndex) ||
+                        (snippetVariableMatcher.start() > invocationArgsEndIndex) ||
+                        (snippetVariableMatcher.end() < invocationArgsStartIndex) ||
+                        (snippetVariableMatcher.end() > invocationArgsEndIndex)) {
+                        // If we found a non-end variable outside of the invocation args, don't change the snippet
+                        if (!END_VARIABLE.equals(snippetVariableMatcher.group())) {
+                            return snippetContent;
+                        }
+                    }
+                }
+
+                // Update the snippet should be updated so that only $0 ends up between the invocation parens
+                String beforeInvocationArgs = snippetContent.substring(0, invocationArgsStartIndex + 1);
+                String afterInvocationArgs = snippetContent.substring(invocationArgsEndIndex - 1);
+                snippetContent = beforeInvocationArgs.replace(END_VARIABLE, "") + END_VARIABLE + afterInvocationArgs.replace(END_VARIABLE, "");
+            }
+        }
+
+        return snippetContent;
+    }
+
+    @Contract("null -> false")
```

Comment:
I added `@Contract` annotations here and below because the IDE's nullability inspection was having trouble determining the relationships between the params and the results and was raising false inspection positives.